dashboard/app/models/concerns/curriculum/course_types.rb

```diff
@@ -15,6 +15,28 @@ module Curriculum::CourseTypes
     validates :participant_audience, acceptance: {accept: SharedCourseConstants::PARTICIPANT_AUDIENCE.to_h.values, message: 'must be facilitator, teacher, or student'}
 
     validate :cannot_have_same_audiences
+    validate :must_have_same_course_type_as_family
+  end
+
+  # All courses in the same family name must have the save instruction_type, instructor_audience, and participant audience
+  def must_have_same_course_type_as_family
+    all_family_courses = nil
+    family_name = self.family_name
+
+    if (is_a?(Script) && unit_group) || is_a?(UnitGroup)
+      family_name = unit_group.family_name if is_a?(Script) && unit_group
+      return if family_name.nil_or_empty?
+      all_family_courses = UnitGroup.all.select {|c| c.family_name == family_name}
```

Comment:
please use `all_courses` instead of `UnitGroup.all`, to avoid hitting the database